velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -717,63 +706,71 @@ uint32_t HiveDataSink::appendWriter(const HiveWriterId& id) {
         writerInfo_.back()->nonReclaimableSectionHolder.get();
   }
 
-  if (options->defaultMemoryReclaimerFactory == nullptr ||
-      options->defaultMemoryReclaimerFactory() == nullptr) {
-    options->defaultMemoryReclaimerFactory = []() {
+  if (options->memoryReclaimerFactory == nullptr ||
+      options->memoryReclaimerFactory() == nullptr) {
+    options->memoryReclaimerFactory = []() {
       return exec::MemoryReclaimer::create();
     };
   }
 
-  if (!options->maxStripeSize) {
-    options->maxStripeSize = std::optional(
-        hiveConfig_->orcWriterMaxStripeSize(connectorSessionProperties));
+  if (insertTableHandle_->compressionKind().has_value()) {
+    options->configs.emplace(
+        velox::dwrf::Config::COMPRESSION.key,
+        std::to_string(insertTableHandle_->compressionKind().value()));
   }
 
-  if (!options->maxDictionaryMemory) {
-    options->maxDictionaryMemory = std::optional(
-        hiveConfig_->orcWriterMaxDictionaryMemory(connectorSessionProperties));
-  }
+  options->configs.emplace(
+      velox::dwrf::Config::STRIPE_SIZE.key,
+      std::to_string(
+          hiveConfig_->orcWriterMaxStripeSize(connectorSessionProperties)));
 
-  if (!options->orcWriterIntegerDictionaryEncodingEnabled) {
-    options->orcWriterIntegerDictionaryEncodingEnabled =
-        hiveConfig_->isOrcWriterIntegerDictionaryEncodingEnabled(
-            connectorSessionProperties);
-  }
+  options->configs.emplace(
+      velox::dwrf::Config::COMPRESSION_BLOCK_SIZE_MIN.key,
+      std::to_string(hiveConfig_->orcWriterMinCompressionSize(
+          connectorSessionProperties)));
 
-  if (!options->orcWriterStringDictionaryEncodingEnabled) {
-    options->orcWriterStringDictionaryEncodingEnabled =
-        hiveConfig_->isOrcWriterStringDictionaryEncodingEnabled(
-            connectorSessionProperties);
-  }
+  options->configs.emplace(
+      velox::dwrf::Config::LINEAR_STRIPE_SIZE_HEURISTICS.key,
+      std::to_string(hiveConfig_->orcWriterLinearStripeSizeHeuristics(
+          connectorSessionProperties)));
 
-  if (!options->orcMinCompressionSize) {
-    options->orcMinCompressionSize = std::optional(
-        hiveConfig_->orcWriterMinCompressionSize(connectorSessionProperties));
-  }
+  options->configs.emplace(
```

Comment:
Shall we move this to dwrf processConfigs callback? We can access things from config in hiveConfig_? thanks!